Bluetooth: controller: Add support for the nRF52810

cvinayak opened the pull request:
The nRF52810 is a low-cost variant of the nRF52832, with a reduced set
of peripherals and memory. This commit adds Bluetooth controller support
for it.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Add support for the nRF52810

The nRF52810 is a low-cost variant of the nRF52832, with a reduced set
of peripherals and memory. This commit adds Bluetooth controller support
for it.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>
Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented on subsys/bluetooth/controller/hal/nrf5/radio/radio_nrf52810.h:
```diff
@@ -0,0 +1,358 @@
+		} RAMPRI;
+	} volatile *NRF_AMLI = (void volatile *)0x40000000UL;
+
+	NRF_AMLI->RAMPRI.CPU0    = 0xFFFFFFFFUL;
```
Can we use the nrfx radio hal for this? If not then I think we should add it in an upcoming version of nrfx, @anangl?

anangl replied:
No, the nrfx RADIO HAL is only for accessing the RADIO peripheral.

carlescufi replied:
OK, thanks. In that case can you make a note to provide nrfx hal functions for AMLI in the future?

MariuszSkamra commented:
I've tested peripheral_dis && eddystone apps and those are working.
btshell and mesh_shell applications do not compile because app exceeds RAM and/or FLASH.
I see that CI failed for the same reason. I think that nrf52810_pca10040 should be added to platform_exclude section in sample.yml/testcase.yml to make the CI happy.

ioannisg reviewed (approved):
LGTM.
Tested the peripheral sample, it works fine.

carlescufi commented:
@MariuszSkamra @cvinayak 
>I think that nrf52810_pca10040 should be added to platform_exclude section in sample.yml/testcase.yml to make the CI happy.

I agree, let's do that to fix CI.

jhedberg commented:
> @jhedberg agreed? or do you want to optimize those samples to make them fit?

@carlescufi no, I agree. Let's exclude them (at least for now).

cvinayak committed:
Bluetooth: tests: exclude btshell/mesh_shell on NRF52810_PCA10040

Exclude building the btshell and mesh_shell application for
NRF52810_PCA10040 boards due to limited RAM and Flash
availability on the nRF52810 SoC.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

MariuszSkamra commented:
@cvinayak I don't think that excluding btshell/mesh_shell is enough. I saw that failed some of the tests/ as well (not related to Bluetooth).

carlescufi commented:
Looks good @cvinayak

carlescufi commented:
@jhedberg @MariuszSkamra we are not enabling Bluetooth by default on this chip. The reason is that if we do, and since this supports an MPU and userspace, the kernel tests will not fit in RAM. I do not think it's a good idea to enable Bluetooth by default on any of our boards anyway, this should really be in the project's `prj.conf`.

carlescufi commented:
@MariuszSkamra you might want to add `CONFIG_ARM_MPU_NRF52X=n` in your `prj.conf` to save more RAM and flash if you don't need the MPU.
